src/Connectors/src/Connectors/CosmosDb/CosmosDbServiceCollectionExtensions.cs

```diff
@@ -73,7 +98,11 @@ public static IServiceCollection AddCosmosDb(this IServiceCollection services, I
         string hostName = GetHostNameFromConnectionString(connectorShim.Options.ConnectionString);
         var logger = serviceProvider.GetRequiredService<ILogger<CosmosDbHealthContributor>>();
 
-        return new CosmosDbHealthContributor(cosmosClient, $"CosmosDB-{serviceBindingName}", hostName, logger);
+        return new CosmosDbHealthContributor(cosmosClient, hostName, logger)
+        {
+            ServiceName = serviceBindingName,
+            Timeout = TimeSpan.FromSeconds(5)
```

Comment:
Is there a simple way we can make this timeout configurable?

Reply:
This is a tricky one. The CosmosDB driver provides `options.RequestTimeout`, but no timeout setting for _establishing_ the socket connection. I found it fails on my machine after 30 seconds with a `SocketException` when the host is unreachable. The CosmosDB driver just passes `CancellationToken.None` to the socket API. I can dig deeper to find where the 30 seconds comes from.

Also, `ReadAccountAsync` doesn't take a `CancellationToken`, so we're stuck with the 30 seconds _when the host is unreachable_. Otherwise, the `options.RequestTimeout` is applied, which users can already set as demonstrated [here](https://github.com/SteeltoeOSS/Samples/blob/latest/Connectors/src/CosmosDb/Program.cs#L20).

So the best we can do is wait for the returned `Task` with our own time-out. However, that doesn't terminate the task itself and is merely a hack. All the other drivers do not have this problem, which is why only the CosmosDB health contributor has this `Timeout` property.

Exposing the property may lead users to believe this is _the_ timeout being used. But that's not the case: if `options.RequestTimeout` has a smaller value, or if `CosmosDbHealthContributor.Timeout` is higher than 30 seconds, then this property has no effect.

Perhaps the timeout is a theoretical case. It is far more likely that an Azure instance is responding slowly, than that its host (something.azure.com) is unreachable.

I wouldn't mind changing the value from 5 seconds to something else (or removing it entirely), but I don't think we should expose it.

Reply:
I think I'm OK with it as-is given this explanation. I was thinking we would call it something like `HealthContributorTimeout` if it was exposed so that it wouldn't be confusing, but it should also be fine this way for now. feel free to resolve this one if no further discussion is desired

Reply:
Ok, I'm resolving this one for now, but thanks for bringing it up. We may need to revisit this if/when adopting the community ASP.NET health checks.